### PR TITLE
Add New Challenge Dictionary App 

### PR DESCRIPTION
**Pull request by satyamsharmasam**
# Title and Issue number 

Title :

Issue No. :

Code Stack : 

Close #<issue_no>

# Checklist:

- [] I have mentioned the issue number in my Pull Request.
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have created a helpful and easy to understand `README.md`
- [] I have updated the Index.html file for my contribution

### CODE REVIEW

**Comment by satyamsharmasam** [comment]
Add new JS challenge Dictionary App

File: docs/javascript/src/challenges/Dictionary App/style.css [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+*{
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+}
+body{
+    width: 100%;
+    font-family: poppins;
+    background-color: rgb(115, 136, 254);
+}
```
Comment (sadanandpai) [inline docs/javascript/src/challenges/Dictionary App/style.css:11]: plz don't add above styles as they get covered through core files

File: docs/javascript/src/challenges/Dictionary App/index.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Dictionary App</title>
+    <link rel="stylesheet" href="style.css">
+    <script src="./script.js" defer></script>
+    <script src="https://kit.fontawesome.com/48486e09d0.js" crossorigin="anonymous"></script>
```
Comment (sadanandpai) [inline docs/javascript/src/challenges/Dictionary App/index.html:9]: plz use only required tags

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by sadanandpai** [comment]
@satyamsharmasam request you to go through the [contribution guidelines](https://github.com/sadanandpai/frontend-mini-challenges/blob/main/CONTRIBUTING.md)

File: docs/javascript/src/challenges/Dictionary App/script.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+const soundBtn = document.querySelector('.sound-btn')
+const wordBody = document.querySelector('.content-body')
+const notFound = document.querySelector('.not-found')
+const audio = document.querySelector('audio')
+
+// fetch data from api and set in our project
+showData = async (word) => {
+    const url = `https://api.dictionaryapi.dev/api/v2/entries/en/${word}`
+    const response = await fetch(url)
+    const word_data = await response.json()
```
Comment (sadanandpai) [inline docs/javascript/src/challenges/Dictionary App/script.js:17]: plz move the fetching part to a different function & also take care of error handling.

File: docs/javascript/src/challenges/Dictionary App/script.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    if (!word_data[0]) {
+        notFound.style.display = 'flex'
+        wordBody.style.display = 'none'
+        return
+    }
+
+    notFound.style.display = 'none'
+    wordBody.style.display = 'flex'
+
```
Comment (sadanandpai) [inline docs/javascript/src/challenges/Dictionary App/script.js:28]: plz provide the styles with class name toggling

File: docs/javascript/src/challenges/Dictionary App/script.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    word_searched.innerText = word_data[0].word
+    partOfSpeech.innerText = word_data[0].meanings[0].partOfSpeech
+    phonetic.innerText = word_data[0].phonetic ? word_data[0].phonetic : ''
+    meaning.innerText = word_data[0].meanings[0].definitions[0].definition
+    sentence.innerText = word_data[0].meanings[0].definitions[0]?.example ? word_data[0].meanings[0].definitions[0]?.example : '';
+    audio.src = word_data[0].phonetics[0].audio || word_data[0].phonetics[1].audio || word_data[0].phonetics[2].audio
```
Comment (sadanandpai) [inline docs/javascript/src/challenges/Dictionary App/script.js:35]: move the above logic to its own function

**Review by sadanandpai — CHANGES_REQUESTED** [review]

File: docs/javascript/src/challenges/Dictionary App/script.js (hunk 5 above)
Comment (satyamsharmasam) [reply docs/javascript/src/challenges/Dictionary App/script.js:35]: okay , i will work on all of these mistakes

**Comment by sadanandpai** [comment]
hi @satyamsharmasam 
plz follow the contribution guidelines.[contribution guidelines](https://github.com/sadanandpai/frontend-mini-challenges/blob/main/CONTRIBUTING.md)

Also plz address the comments

**Pull request closed without merging**